net/tcp/tcp_monitor.c

```diff
@@ -346,15 +346,13 @@ void tcp_close_monitor(FAR struct socket *psock)
   /* Find and free the the connection event callback */
 
   net_lock();
-  for (cb = conn->connevents;
-       cb != NULL && cb->priv != (FAR void *)psock;
-       cb = cb->nxtconn)
-    {
-    }
 
-  if (cb != NULL)
+  for (cb = conn->connevents; cb != NULL; cb = cb->nxtconn)
     {
-      devif_conn_callback_free(conn->dev, cb, &conn->connevents);
+      if (cb->priv == (FAR void *)psock)
+        {
+          devif_conn_callback_free(conn->dev, cb, &conn->connevents);
```

Comment:
doesn't this put the "cb" onto the freelist, using cb->nxtconn, and thus break the iteration?

Reply:
see the Summary

Reply:
i read the summary as of today but it doesn't seem answering my concern about list corruption.

Reply:
Yes, you are right, this fix doesn't works.
I will abandon this RP until find the root case.